### PR TITLE
Change ownership and upgrade to openzeppelin v2

### PR DESCRIPTION
**Pull request by leonprou**
- token owner will be the owner of the token on home
- upgrading to open zeppelin v2 for more clean separation of roles

### CODE REVIEW

File: contracts/ERC677BridgeToken.sol [hunk 1]
```diff
@@ -49,14 +51,19 @@ contract ERC677BridgeToken is
     }
 
     function getTokenInterfacesVersion() public pure returns(uint64 major, uint64 minor, uint64 patch) {
-        return (2, 0, 0);
+        return (3, 0, 0);
```
Comment (leonprou) [inline contracts/ERC677BridgeToken.sol:54]: The changes I made are not essentional but do change the exposed API, so by semver this requests for major change

File: contracts/upgradeable_contracts/factories/BasicBridgeFactory.sol [hunk 2]
```diff
@@ -6,7 +6,7 @@ import "../EternalOwnable.sol";
 contract BasicBridgeFactory is EternalStorage, EternalOwnable {
 
     function getBridgeFactoryVersion() public pure returns(uint64 major, uint64 minor, uint64 patch) {
-        return (2, 2, 0);
+        return (3, 2, 0);
```
Comment (leonprou) [inline contracts/upgradeable_contracts/factories/BasicBridgeFactory.sol]: adding `_tokenOwner` arg to `deployHomeBridge`

**Comment by LiorRabin** [comment]
Nice! looks better without all the manual `add` & `sub` calls :) 
I think you can remove the commented imports.

**Review by LiorRabin — APPROVED** [review]

File: contracts/ERC677BridgeToken.sol [hunk 3]
```diff
@@ -24,7 +26,7 @@ contract ERC677BridgeToken is
         string _name,
         string _symbol,
         uint8 _decimals)
-    public DetailedERC20(_name, _symbol, _decimals) {}
+    public ERC20Detailed(_name, _symbol, _decimals) {}
 
     function setBridgeContract(address _bridgeContract) onlyOwner public {
```
Comment (leonprou) [inline contracts/ERC677BridgeToken.sol]: The token owner can change bridge contract. Changing the bridge to other contract will break the bridge.

File: contracts/upgradeable_contracts/factories/BasicBridgeFactory.sol (hunk 2 above)
Comment (bejavu) [reply contracts/upgradeable_contracts/factories/BasicBridgeFactory.sol]: change to 3.0.0

File: contracts/ERC677BridgeToken.sol (hunk 3 above)
Comment (bejavu) [reply contracts/ERC677BridgeToken.sol]: owner should not be allowed to change the bridge, or it is the bridge itself.

**Review by bejavu — CHANGES_REQUESTED** [review]

File: contracts/ERC677BridgeToken.sol (hunk 3 above)
Comment (leonprou) [reply contracts/ERC677BridgeToken.sol]: Makes sense. I make `setBridgeContract` with `onlyMinter` modifier.  The minter is the bridge. This makes the permissions similar to the previous case. Before it was `onlyOwner` when the bridge was the owner.

**Review by bejavu — APPROVED** [review]

**Pull request merged**